src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,574 @@
+                record.emplace_back(std::move(value));
+            }
+            std::vector<cpp2::ColumnValue> row(yields_.size());
+            auto index = 0;
+            for (auto &column : record) {
+                switch (column.which()) {
+                    case VAR_INT64:
+                        row[index].set_integer(boost::get<int64_t>(column));
+                        break;
+                    case VAR_DOUBLE:
+                        row[index].set_double_precision(boost::get<double>(column));
+                        break;
+                    case VAR_BOOL:
+                        row[index].set_bool_val(boost::get<bool>(column));
+                        break;
+                    case VAR_STR:
+                        row[index].set_str(boost::get<std::string>(column));
+                        break;
+                    default:
+                        LOG(FATAL) << "Unknown VariantType: " << column.which();
+                }
+                index++;
+            }
+            rows.emplace_back();
+            rows.back().set_columns(std::move(row));
+        }
+    }
+    resp_->set_rows(std::move(rows));
+    DCHECK(onFinish_);
+    onFinish_();
+}
+
+
+void UpdateEdgeExecutor::setupResponse(cpp2::ExecutionResponse &resp) {
+    if (resp_ == nullptr) {
+        resp_ = std::make_unique<cpp2::ExecutionResponse>();
+    }
+    resp = std::move(*resp_);
+}
+
+
+void UpdateEdgeExecutor::fetchDstVertexProps() {
+    auto spaceId = ectx()->rctx()->session()->space();
+    auto statusOr = getDstProps();
+    if (!statusOr.ok()) {
+        auto msg = "Internal error about getDstProps.";
+        LOG(ERROR) << msg;
+        onError_(Status::Error(msg));
+        return;
+    }
+    auto returns = statusOr.value();
+    if (returns.empty()) {
+        return;
+    }
+    folly::Baton<true, std::atomic> baton;
+    std::vector<VertexID> vertices;
+    vertices.emplace_back(edge_.get_dst());
+    auto future = ectx()->storage()->getVertexProps(spaceId,
+                                                    std::move(vertices),
+                                                    std::move(returns));
+    auto *runner = ectx()->rctx()->runner();
+    auto cb = [this, &baton] (auto &&result) {
+        auto completeness = result.completeness();
+        if (completeness != 100) {
+            auto msg = "Internal error when fetch dst vertex props.";
+            LOG(ERROR) << msg;
+            onError_(Status::Error(msg));
+            return;
+        }
+        if (this->vertexHolder_ == nullptr) {
+            this->vertexHolder_ = std::make_unique<VertexHolder>();
+        }
+        for (auto& resp : result.responses()) {
+            this->vertexHolder_->add(resp);
+        }
+        baton.post();
+    };
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        DCHECK(onError_);
+        onError_(Status::Error("Internal error: getVertexProps!"));
```

Comment:
maybe here can not call onError_, because onError_ will free the point "this", so if this thread wake up from baton.wait, the memory of "this" has been free.

Reply:
fixed!